src/syntax/ast.rs

```diff
@@ -1821,7 +1843,15 @@ impl ForLoop {
 
     /// The expression to iterate over.
     pub fn iter(&self) -> Expr {
-        self.0.cast_first_match().unwrap_or_default()
+        match self.pattern() {
+            Pattern::Ident(_) => self
+                .0
+                .children()
+                .filter_map(SyntaxNode::cast)
+                .nth(1)
+                .unwrap_or_default(),
+            Pattern::Destructuring(_) => self.0.cast_first_match().unwrap_or_default(),
+        }
```

Comment:
here you could also skip_while you haven't found `SyntaxKind::In` and then cast the first expression after